pywr/parameters/_thresholds.pyx

```diff
@@ -187,6 +187,98 @@ cdef class NodeThresholdParameter(AbstractThresholdParameter):
 NodeThresholdParameter.register()
 
 
+cdef class MultipleThresholdIndexParameter(IndexParameter):
+    """ Multiple threshold holder returning an index based on the previous flow in a node
+    against the given thresholds.
+
+    Parameters
+    ----------
+    node : `pywr.core.AbstractNode`
+    thresholds : iterable of `Parameter` instances or floats
+    """
+    def __init__(self, model, node, thresholds, use_max_flow=False, **kwargs):
+        super(MultipleThresholdIndexParameter, self).__init__(model, **kwargs)
+        self.node = node
+        self.thresholds = thresholds
+
+    cpdef int index(self, Timestep timestep, ScenarioIndex scenario_index) except? -1:
+        """Returns the index of the first threshold the node flow is above
+
+        The index is zero-based. For example, if only one threshold is
+        supplied then the index is either 0 (above) or 1 (below). For two
+        thresholds the index is either 0 (above both), 1 (in between), or 2 (below
+        both), and so on.
+        """
+        cdef double flow
+        cdef int index, j
+
+        flow = self.node._prev_flow[scenario_index.global_id]
+        index = len(self.thresholds)
+        for j, threshold in enumerate(self.thresholds):
+            if isinstance(threshold, (float, int)):
```

Comment:
I think the control curve equivalent of this turns floats and ints into a `ConstantParameter` on load/creation. At the moment this loop won't Cythonise too well because `threshold` can not be cdef'd. If you knew they were all `Parameter` instances then it could be `cdef Parameter` and probably be quicker.

Reply:
see comment above